packages/eslint-plugin/src/rules/prefer-nullish-coalescing.ts

```diff
@@ -475,6 +482,12 @@ export default createRule<Options, MessageIds>({
             return true;
           }
 
+          const tsNode = parserServices.esTreeNodeToTSNodeMap.get(
+            nullishCoalescingLeftNode,
+          );
+          const type = checker.getTypeAtLocation(tsNode);
```

Comment:
[Refactor] Nit: a nicer new API for this ~

```suggestion
          const type = parserServices.getTypeAtLocation(
            nullishCoalescingLeftNode,
          );
```

This is existing code, so definitely not a blocker 😄

Reply:
Nice!